zilliqa/src/pool.rs

```diff
@@ -127,6 +133,60 @@ impl TransactionPool {
         }
     }
 
+    pub fn preview_content(&self) -> TxPoolContent {
+        // First make a copy of 'ready' transactions
+        let mut ready: BinaryHeap<ReadyItem> = BinaryHeap::new();
+        for item in self.ready.iter() {
+            ready.push(*item);
+        }
+
+        let mut pending = Vec::new();
+        let mut pending_set = HashSet::new();
+
+        // Find all transactions that are pending for inclusion in the next block
+        while let Some(ReadyItem { tx_index, .. }) = ready.pop() {
+            // Should we care about nonceless txns?
+            if let TxIndex::Intershard(_, _) = tx_index {
+                continue;
+            }
+
+            // Expired txn?
+            let Some(txn) = self.transactions.get(&tx_index) else {
+                continue;
+            };
+
+            if pending_set.contains(&txn.hash) {
+                continue;
+            }
+
+            pending.push(txn.clone());
+            pending_set.insert(txn.hash);
+
+            let Some(next) = tx_index.next() else {
+                continue;
+            };
+
+            if let Some(next_txn) = self.transactions.get(&next) {
+                ready.push(next_txn.into());
+            }
```

Comment:
Is it worth de-duplicating this logic with the same logic in `fn best_transaction`? Or is it just superficially similar and actually different? I ask because it took me a second to verify this was maintaining the heap in the same way.